src/mbgl/renderer/raster_bucket.cpp

```diff
@@ -12,8 +13,7 @@ RasterBucket::RasterBucket(PremultipliedImage&& image_) : image(std::move(image_
 }
 
 void RasterBucket::upload(gl::Context& context) {
-    texture = context.createTexture(image);
-    image = {};
+    texture = context.createTexture(util::unpremultiply(std::move(image)));
```

Comment:
I'll leave it using `unpremultiply` for now. We can revisit if it proves to be an issue. I did move the computation to the worker thread though.